### ISSUE TITLE
introducing new vbank assets results in O(n) work to update smart wallets

### ISSUE THREAD

**Issue by dckc** [Agoric/Agoric#6652]
## Describe the bug

As noted in [a Dec 5 comment on #6625](https://github.com/Agoric/agoric-sdk/issues/6625#issuecomment-1338145992), when [proposal 17 to add DAI](https://wallet.keplr.app/chains/agoric/proposals/17) passed and 2 new vbank issues were added, each smart wallet woke up, added a purse, and published its new state to vstorage:

https://github.com/Agoric/agoric-sdk/blob/2c812d22161cd297587979b262eab6e2cc76e23d/packages/smart-wallet/src/smartWallet.js#L523-L524

This is clearly not consistent with our [Limit work by message size](https://github.com/Agoric/agoric-sdk/wiki/Limit-work-by-message-size) design constraint.

## To Reproduce
1. Add several hundred wallets
2. call `E(vbank).addAsset(...)`
3. see hundreds of purses created and hundreds of vstorage updates

## Expected behavior

_TBD_, but something that respects our [Limit work by message size](https://github.com/Agoric/agoric-sdk/wiki/Limit-work-by-message-size) design constraint.

RPC clients should be able to get the list of vbank brands without relying on updates to individual wallets:
- [x] #5762

design discussion: [IBIS: How to introduce new VBank Issuers to Wallets \- Google Docs](https://docs.google.com/document/d/10VKtO1EfHDdJ8KE2Q_Kaft5pwTCR-I2kK1YK-07YiHg/edit#heading=h.dz8ctp7thx5w)

## Platform Environment

mainnet. pismoA

**Comment by dckc** [comment]
## Failing test

ea697a5e6 cc @turadg @arirubinstein 

```
~/projects/agoric-sdk/packages/smart-wallet$ yarn test test/test-addAsset.js 
...
  ✔ [expected fail] avoid O(wallets) storage writes for a new asset
    ℹ {
        base: {
          wallets: 2,
          writes: 4,
        },
        test: {
          wallets: 8,
          writes: 16,
        },
      }
```

**Comment by dckc** [comment]
I coded up lazy purse creation in #6774 . But another option that's starting to look good is

## Skip chainStorage writes for vbank purse balances

Now that we have `agoricNames.vbankAssets` (#5762 ), clients can just subscribe to queries from `x/bank` and join with `vbankAssets` on `denom` to get brand, displayInfo.decimalPlaces, etc. The smartWallet need not track purse balances; it can just use purses for the duration of handling an offer.

This would update our approach to
 - #6184

It would be a significant change to the wallet front end. And we would need another approach when we add non-vbank assets (LP tokens etc.)

I'm not sure how the REPL wallet API would work either.

**Comment by dckc** [comment]
Let's call this fixed in #5412
and consider #6807 part of #6788
